spring-integration-core/src/main/java/org/springframework/integration/gateway/MessagingGatewaySupport.java

```diff
@@ -333,6 +352,14 @@ private Object doSendAndReceive(Object object, boolean shouldConvert) {
 					error = ((ErrorMessage) reply).getPayload();
 				}
 			}
+			if (reply == null && this.errorOnTimeout) {
```

Comment:
@craignicoll 

Gateways are for two-way interaction (request/reply); this fix corrected a problem where, if a timeout occurred, `200 OK` was returned; which was incorrect.

You should use an `<http:inbound-channel-adapter/>` if you never expect a reply (such as in your case). Channel adapters are used for one-way interactions with the message flow (although in this case, a 200 OK is sent to the caller). One-way in this context means the flow doesn't return a reply.

In fact, this fix has simply exposed a bug in your configuration; before this change, the thread would be suspended for 1 second (the default `reply-timeout`) after the flow completes (waiting for a reply). This means that all your requests take 1 second longer than expected.